test/new-e2e/tests/npm/ec2_1host_test.go

```diff
@@ -0,0 +1,166 @@
+		e2eParams = []params.Option{params.WithDevMode(), params.WithSkipDeleteOnFailure()}
+		s.DevMode = true
+	}
+
+	// Source of our kitchen CI images test/kitchen/platforms.json
+	// Other VM image can be used, our kitchen CI images test/kitchen/platforms.json
+	// ec2params.WithImageName("ami-a4dc46db", os.AMD64Arch, ec2os.AmazonLinuxOS) // ubuntu-16-04-4.4
+	e2e.Run(t, s, e2e.FakeIntakeStackDef(nil, agentparams.WithSystemProbeConfig(NPMsystemProbeConfig)), e2eParams...)
+}
+
+// TestFakeIntakeNPM Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMSuite) TestFakeIntakeNPM() {
+	t := v.T()
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.DevMode {
+		v.Env().Fakeintake.FlushServerAndResetAggregators()
+	}
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate a connection
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+
+		hostnameNetID, err := v.Env().Fakeintake.GetConnectionsNames()
+		assert.NoError(c, err, "GetConnectionsNames() errors")
+		assert.NotZero(c, len(hostnameNetID), "no connections yet")
+		targetHostnameNetID = hostnameNetID[0]
+
+		t.Logf("hostname+networkID %v seen connections", hostnameNetID)
+	}, 60*time.Second, time.Second, "no connections received")
+
+	// looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		cnx, err := v.Env().Fakeintake.GetConnections()
+		assert.NoError(t, err)
+
+		assert.Greater(c, len(cnx.GetPayloadsByName(targetHostnameNetID)), 2, "not enough payloads")
+		if len(cnx.GetPayloadsByName(targetHostnameNetID)) < 3 {
+			return
+		}
+
+		var payloadsTimestamps []time.Time
+		for _, cc := range cnx.GetPayloadsByName(targetHostnameNetID) {
+			payloadsTimestamps = append(payloadsTimestamps, cc.GetCollectedTime())
+		}
+		dt := float64(payloadsTimestamps[2].Sub(payloadsTimestamps[1])) / float64(time.Second)
+		t.Logf("hostname+networkID %v diff time %f seconds", targetHostnameNetID, dt)
+
+		// we want the test fail now, not retrying on the next payloads
+		if math.Abs(dt-30) > 0.5 {
+			t.Log("delta between collection is higher that 500ms")
+			c.FailNow()
+		}
+	}, 90*time.Second, time.Second, "no connections received")
+}
+
+// TestFakeIntakeNPM_TCP_UDP_DNS validate we received tcp, udp, and DNS connections
+// with some basic checks, like IPs/Ports present, DNS query has been captured, ...
+func (v *ec2VMSuite) TestFakeIntakeNPM_TCP_UDP_DNS() {
+	t := v.T()
+
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate connections
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
+		v.Env().VM.Execute("dig @8.8.8.8 www.google.ch")
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Instead of triggering again the action, I would rather first wait for `system-probe` to be ready, with a loop that also retries the actions. Out of that, I would trigger the action once and retry the assertion 

I recently wrote this doc on handling flakiness in 
 e2e tests https://datadoghq.atlassian.net/wiki/spaces/AETET/pages/3182362690/Handle+flakiness+in+e2e+tests